docs/doxygen/logging/logging.dox

```diff
@@ -0,0 +1,44 @@
+/**
+@page logging Logging Infrastructure
+@brief This page contains an overview of the logging infrastructure used by the C-SDK Libraries.
+
+<p>
+The C-SDK libraries utilize one of the following 4 logging macro interfaces, each with a different verbosity level:
+- `LogError` for **Error** level
+- `LogWarn` for **Warning** level
+- `LogInfo` for **Info** level
+- `LogDebug` for **Debug** level
+
+**Note: By default, each library contains empty definitions for the above logging macros, thereby generating no code for logging calls.**
+
+To enable logging within the libraries, the customer application should implement these macros through the library-specific config file (for example, the `core_mqtt_config.h` file for MQTT library).
+The implementation of these macros can be provided as a mapping to the customer's platform-specific logging framework.
+</p>
+
+@section design Design Architecture
+@brief The logging infrastructure architecture follows a dependency inversion design principle of allowing the customer application to inject an implementation of logging interface macros through the library-specific config file (like `core_mqtt_config.h`).
+
+@image html logging_architecture.png width=40%
```

Comment:
How was this image created? Will the source also be checked into version control?

Reply:
Sure, will add it